lib/class-wp-rest-settings-controller.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+/**
+ * Manage a WordPress site's settings.
+ */
+
+class WP_REST_Settings_Controller extends WP_REST_Controller {
+	public function __construct() {
+		$this->namespace = 'wp/v2';
+		$this->rest_base = 'settings';
+	}
+
+	/**
+	 * Register the routes for the objects of the controller.
+	 */
+	public function register_routes() {
+		register_rest_route( $this->namespace, '/' . $this->rest_base, array(
+			array(
+				'methods'          => WP_REST_Server::READABLE,
+				'callback'         => array( $this, 'get_item' ),
+				'args'             => array(),
+				'permission_callback' => array( $this, 'get_item_permissions_check' ),
+			),
+			array(
+				'methods'          => WP_REST_Server::EDITABLE,
+				'callback'         => array( $this, 'update_item' ),
+				'args'             => $this->get_endpoint_args_for_item_schema( WP_REST_Server::EDITABLE ),
+				'permission_callback' => array( $this, 'get_item_permissions_check' ),
+			),
+			'schema' => array( $this, 'get_public_item_schema' ),
+		) );
+	}
+
+	public function get_item_permissions_check( $request ) {
+		return current_user_can( 'manage_options' );
+	}
+
+	/**
+	 * Get the settings.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|array
+	 */
+	public function get_item( $request ) {
+		$options  = $this->get_registered_options();
+		$response = array();
+
+		foreach ( $options as $name => $args ) {
+			// default to a null value as "null" in the response means "not set"
+			$response[ $name ] = get_option( $args['option_name'], $args['schema']['default'] );
+
+			// because get_option() is lossy, we have to
+			// cast values to the type they are registered with.
+			switch ( $args['schema']['type'] ) {
+				case 'string':
+					$response[ $name ] = strval( $response[ $name ] );
+					break;
+				case 'number':
+					$response[ $name ] = floatval( $response[ $name ] );
+					break;
+				case 'boolean':
+					$response[ $name ] = (bool) $response[ $name ];
+					break;
+			}
```

Comment:
The contents here should probably be split into a separate method for reusability.